lib/apimatic-core/utilities/api_helper.rb

```diff
@@ -228,19 +226,62 @@ def self.clean_url(url)
       protocol + query + parameters
     end
 
+    # Deserialize the response based on the provided union_type.
+    # @param [UnionType] union_type The union type to validate and deserialize the response.
+    # @param [Object] response The response object to be deserialized.
+    # @param [Boolean] should_deserialize Flag indicating whether the response should be deserialized.
+    #                                    Default is true.
+    # @return [Object] The deserialized response based on the union_type.
+    def self.deserialize_union_type(union_type, response, should_symbolize = false, should_deserialize = false)
+      response = ApiHelper.json_deserialize(response, false, true) if should_deserialize
+
+      union_type_result = union_type.validate(response)
+
+      union_type_result.deserialize(response, should_symbolize: should_symbolize)
+    end
+
+    def self.apply_primitive_converter(value)
+      # Attempt to deserialize as Integer
+      return value.to_i if value.match?(/^\d+$/)
+
+      # Attempt to deserialize as Float
+      return value.to_f if value.match?(/^\d+(\.\d+)?$/)
+
+      # Attempt to deserialize as Boolean
+      return true if value.downcase == 'true'
+      return false if value.downcase == 'false'
+
+      # Default: return the original string
+      value
+    end
+
+    # Checks if a value or all values in a nested structure satisfy a given type condition.
+    # value        - The value or nested structure to be checked.
+    # type_callable - A callable object that defines the type condition.
+    # Returns true if the value or all values in the structure satisfy the type condition, false otherwise.
+    def self.valid_type?(value, type_callable)
+      if value.is_a?(Array)
+        value.all? { |item| valid_type?(item, type_callable) }
+      elsif value.is_a?(Hash)
+        value.values.all? { |item| valid_type?(item, type_callable) }
+      else
+        !value.nil? && type_callable.call(value)
+      end
+    end
+
     # Parses JSON string.
     # @param [String] json A JSON string.
-    # rubocop:disable Style/OptionalBooleanParameter
-    def self.json_deserialize(json, should_symbolize = false)
+    def self.json_deserialize(json, should_symbolize = false, apply_primitive_converter = false)
```

Comment:
Nits, Non-blocker: `apply_primitive_converter` should be renamed to `allow_primitive_type_parsing`. Also further detail should be added as part of the docstring.

Reply:
done